pkg/azure/client.go

```diff
@@ -96,8 +105,31 @@ func (az *azClient) GetPublicIP(resourceID string) (n.PublicIPAddress, error) {
 }
 
 // DeployGateway is a method that deploy the appgw and related resources
-func (az *azClient) DeployGateway(subnetID string) (err error) {
+func (az *azClient) DeployGatewayWithVnet(resourceGroupName ResourceGroup, vnetName ResourceName, subnetName ResourceName, subnetPrefix string) (err error) {
+	vnet, err := az.getVnet(resourceGroupName, vnetName)
+	if err != nil {
+		return
+	}
+
+	glog.Infof("Checking the Vnet %s for a subnet with prefix %s", vnetName, subnetPrefix)
```

Comment:
We could set that to `.V(1)` -- seems useful and will be done once in the life of AGIC, right?